### PR TITLE
Higher zs

### PR DESCRIPTION
**Pull request by ssim**
## Motivation
Trying to make minor "fudge" to Carsus to accommodate higher Z elements. Issue is that NIST source for ground state J-values was returning NaNs.

See also: #168.

## Changes
Added a catch in the NIST ionization code to insert "J=0" in place of NaNs - i.e. adopt ground state stat. weight of 1 if it has not been able to inject a value from NIST.

Also minor edit associated with the setting of matching criteria.

## Description
Changes made in nist/ionization.py
(uses NaN check and replaced NaN with 0)
and in gfall.py 
(included "else" condition for the "unique_level_identifier")

### CODE REVIEW

File: carsus/io/nist/ionization.py [hunk 1]
```diff
@@ -154,6 +154,12 @@ def parse_ground_level(row):
             except KeyError:
                 pass
 
+            # To handle cases where the ground level J has not been understood:
+            # Take as assumption J=0
+            # Consider adding a log or warning message???
+            if (np.isnan(lvl["J"])):
```
Comment (epassaro) [inline carsus/io/nist/ionization.py:160]: I would try to use:

`if lvl["J"].isna():`

to keep the code most Pandas-friendly.

Comment (ssim) [reply carsus/io/nist/ionization.py:160]: Hello - I tried replacing line 160 from the above with

if (lvl["J"].isna()):

instead in my local copy but then it complains as follows:

AttributeError: 'numpy.float64' object has no attribute 'isna'

I guess this is some Pandas thing (I don't know Pandas well). Is there some syntax thing wrong here?

Comment (epassaro) [reply carsus/io/nist/ionization.py:160]: Ok! Then don't worry. I'm approving these changes right now.

File: carsus/io/nist/ionization.py [hunk 2]
```diff
@@ -154,6 +154,12 @@ def parse_ground_level(row):
             except KeyError:
                 pass
 
+            # To handle cases where the ground level J has not been understood:
+            # Take as assumption J=0
+            # Consider adding a log or warning message???
```
Comment (chvogl) [inline carsus/io/nist/ionization.py:159]: I think adding a warning message would be a really good idea. Maybe something along these lines could work:
```
print(f"Set J=0 for ground state of species ({row['atomic_number']}, {row['ion_charge']}).")
```

Comment (epassaro) [reply carsus/io/nist/ionization.py:159]: I forgot about this. IIRC there are 147 missing `J`'s betweet `Pr` to `U`, so the logging would be very long. But I don't see a better solution.

Comment (ssim) [reply carsus/io/nist/ionization.py:159]: Shall I go ahead and add Christian's suggested line, then? I can do that easily enough.

Comment (epassaro) [reply carsus/io/nist/ionization.py:159]: Yes, do it.

Comment (ssim) [reply carsus/io/nist/ionization.py]: Ok - I've added that now - and it seems to work ok! Have re"pushed" - so hopefully that shows up properly here now too! Thanks!

**Comment by ssim** [comment]
Hello @epassaro - I just tried to resolve some conflicts that were here (with help with @jamesgillanders ) - can we merge this in now if all are happy?

**Pull request merged**